### PR TITLE
Fix full horizontal boxes overflowing centered layers

### PR DESCRIPTION
**Pull request by jonathonwalz**
#### What does this PR do?

Fixes an issue where a `<Box>` that has `full` set to `true` or `horizontal` will overflow a `<Layer align="center">`. [Example CodePen](http://codepen.io/jonathonwalz/pen/BporVd), should have a padded red box on both sides, not just the left.

#### What testing has been done on this PR?
Manually tested on the project where the issue was noticed. Tested with various different browser widths from 320px to 2048px.

#### Do the grommet docs need to be updated?
No

#### Should this PR be mentioned in the release notes?
Maybe? Not sure what the conventions are

#### Is this change backwards compatible or is it a breaking change?
Yes, just a minor css change

### CODE REVIEW

File: src/scss/grommet-core/_objects.layer.scss [hunk 1]
```diff
@@ -111,6 +111,14 @@
       max-height: calc(100vh - #{double($inuit-base-spacing-unit)});
       max-width: calc(100vw - #{double($inuit-base-spacing-unit)});
       transform: translate(50%, 50%);
+
```
Comment (alansouzati) [inline src/scss/grommet-core/_objects.layer.scss]: Thanks for your contribution, but I guess this problem is unrelated with the align center aspect of the layer.

I believe the reason for the problem is this line https://github.com/grommet/grommet/blob/master/src/scss/grommet-core/_objects.layer.scss#L61

I'm currently investigating it more.

Comment (alansouzati) [inline src/scss/grommet-core/_objects.layer.scss]: Yeah I just validated it, remove these lines fix the issue. Do you mind to update your pull request to reflect it? Please remove from line 60 to 71.

Comment (jonathonwalz) [inline src/scss/grommet-core/_objects.layer.scss]: Could be, I haven't dug much into the inner workings of grommet. In my testing in the codepen, I only saw the issue with center and not the other alignments. I think because the other alignments don't add any left/right spacing outside the layer. (Maybe that's also a bug?)

Comment (jonathonwalz) [inline src/scss/grommet-core/_objects.layer.scss]: Can do. I'll get that done later today. Thanks for looking into it!

Comment (alansouzati) [inline src/scss/grommet-core/_objects.layer.scss]: Thanks you for the contribution!

Comment (jonathonwalz) [inline src/scss/grommet-core/_objects.layer.scss]: @alansouzati Ok, I've switched to deleting those lines. I can also confirm removing them fixes the issue in my application as well. Thank you!

**Comment by alansouzati** [comment]
Thanks, welcome to the team!
